packages/rainbowkit/src/components/RainbowKitProvider/ModalContext.tsx

```diff
@@ -119,13 +122,18 @@ export function ModalProvider({ children }: ModalProviderProps) {
           openChainModal:
             connectionStatus === 'connected' ? openChainModal : undefined,
           openConnectModal:
-            connectionStatus === 'disconnected' ||
-            connectionStatus === 'unauthenticated'
+            // Prevent opening the connect modal during disconnecting mode. Even though `connectionStatus`
+            // may mark it as 'disconnected', we don't want the user to open the modal as the wagmi state
+            // still considers it 'connected'
+            !isDisconnecting &&
+            (connectionStatus === 'disconnected' ||
+              connectionStatus === 'unauthenticated')
               ? openConnectModal
```

Comment:
You technically can still open the connect modal if your `connectionStatus` is returning `disconnected`. 

If we allow users to open the connect modal by the time WalletConnect takes time to disconnect, they'll face the issue with shim disconnection which is a known issue in wagmi rn. Ofc only if they start connecting different wallets at once, but we'll make sure to disable it.